joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/GoSrc2Cpg.scala

```diff
@@ -22,31 +22,36 @@ import io.shiftleft.codepropertygraph.generated.Languages
 import java.nio.file.Paths
 import scala.util.Try
 
-class GoSrc2Cpg extends X2CpgFrontend[Config] {
+class GoSrc2Cpg(goGlobalOption: Option[GoGlobal] = Some(GoGlobal())) extends X2CpgFrontend[Config] {
   private val report: Report = new Report()
 
+  private var goMod: Option[GoModHelper] = None
   def createCpg(config: Config): Try[Cpg] = {
     withNewEmptyCpg(config.outputPath, config) { (cpg, config) =>
       File.usingTemporaryDirectory("gosrc2cpgOut") { tmpDir =>
-        val goGlobal = GoGlobal()
+        val goGlobal = goGlobalOption.getOrElse(GoGlobal())
```

Comment:
This might be cleaner
```suggestion
        goGlobalOption.getOrElse(GoGlobal()).foreach { goGlobal =>
```

Reply:
@DavidBakerEffendi following will work `goGlobalOption.foreach { goGlobal => ... }`  but that will not help. I can't call `.foreach` after  call `getOrElse(GoGlobal())`

Reply:
done